packages/core/src/api/nodeConversions/nodeConversions.ts

```diff
@@ -182,37 +235,128 @@ function contentNodeToInlineContent(contentNode: Node) {
       }
     }
 
-    if (linkMark && currentLink && linkMark.attrs.href === currentLink.href) {
-      // if the node is a link that matches the current link, add it to the current link
-      currentLink.content.push({
-        type: "text",
-        text: node.textContent,
-        styles,
-      });
-    } else if (linkMark) {
-      // if the node is a link that doesn't match the current link, create a new link
-      currentLink = {
-        type: "link",
-        href: linkMark.attrs.href,
-        content: [
-          {
+    // Parsing links and text.
```

Comment:
can you explain these changes? e.g.: why are so many changes necessary?

Reply:
Since I had to change `currentLink` to `currentContent`, which can be either a link or styled text, I had to add more conditions to the existing code, which ended up being really difficult to read and understand.